File: packages/zynth-bluetooth/ios/ZynthBluetooth.swift
import Foundation
import ZynthKit

@objc(ZynthBluetooth)
public final class ZynthBluetooth: NSObject {
  private static var module: BluetoothModule?

  @objc public static func initialize(with runtime: ZynthRuntime) {
    guard module == nil else {
      return
    }

    let instance = BluetoothModule(runtime: runtime)
    runtime.installModules([instance])
    module = instance
  }

  @objc public static func cleanup() {
    module = nil
  }
}

File: packages/zynth-bluetooth/android/ZynthBluetooth/src/main/java/dev/zynth/bluetooth/ZynthBluetooth.kt
package dev.zynth.bluetooth

import androidx.activity.ComponentActivity
import com.zynth.kit.runtime.ZynthRuntime

object ZynthBluetooth {
  @JvmStatic
  fun initialize(activity: ComponentActivity, runtime: ZynthRuntime) {
    val module = BluetoothModule(activity, runtime)
    module.setupPermissionLauncher()
    runtime.installModules(listOf(module))
  }
}
